scripts/genVis.py

```diff
@@ -1,58 +1,140 @@
+import argparse
+import itertools
 
-parser.add_option('-i','--inFile',default='',type=str,
-                  help='Specify input expression matrix file name')
+# Define arguments
+parser = argparse.ArgumentParser("Visualize the BoolODE simulation of cells.")
+parser.add_argument('-f', '--pathToFiles', default='', type=str, help='Specify path to files.')
+parser.add_argument('-p', '--pca', nargs='*', help='Call to visualize with PCA. Specify dimension (2 or 3) as argument.'
+                                                   'Default is 2.')
+parser.add_argument('-t', '--tsne', nargs='*', help='Call to visualize with t-SNE. '
+                                                    'Specify dimension (2 or 3) as argument. Default is 2.')
+parser.add_argument('-u', '--umap', nargs='*', help='Call to visualize with UMAP. '
+                                                    'Specify dimension (2 or 3) as argument. Default is 2.')
+parser.add_argument('-c', '--handleExternalClusterFile', action='store_true', default=False,
+                    help='Call to handle an external cluster file.')
+parser.add_argument('-n', '--plotTitle', default='', nargs='*', help='Write a plot title.')
 
-parser.add_option('-p','--pseudoTimeFile',default='',type=str,
-                  help='Specify pseudoTimeFile file name')
+# Parse arguments
+args = parser.parse_args()
+path = args.pathToFiles
+inFile = path + "/expressiondata.csv"
+timeFile = path + "/pseudotime.csv"
+pca_flag = args.pca is not None
+tsne_flag = args.tsne is not None
+umap_flag = args.umap is not None
+clusterFile = args.pathToFiles + "/clusterids.csv"
 
-parser.add_option('-t','--tsne',action='store_true',default=False,
-                  help='Visualized tsne instead of default PCA')
+# Do PCA, tSNE, UMAP
+DF = pd.read_csv(inFile, sep=',', index_col=0)
+Cells = DF.T.values
+DRDF = pd.DataFrame(index=pd.Index(list(DF.columns)))
+if pca_flag:
+    if len(args.pca) == 0:
+        pca_dim = 2
+    else:
+        pca_dim = int(args.pca[0])
+    pca_embed = PCA(n_components=pca_dim).fit_transform(Cells)
+    DRDF['PCA1'] = pca_embed[:, 0]
+    DRDF['PCA2'] = pca_embed[:, 1]
+    if pca_dim == 3:
+        DRDF['PCA3'] = pca_embed[:, 2]
+if tsne_flag:
+    if len(args.tsne) == 0:
+        tsne_dim = 2
+    else:
+        tsne_dim = int(args.tsne[0])
+    tsne_embed = TSNE(n_components=tsne_dim).fit_transform(Cells)
+    DRDF['tsne1'] = tsne_embed[:, 0]
+    DRDF['tsne2'] = tsne_embed[:, 1]
+    if tsne_dim == 3:
+        DRDF['tsne3'] = tsne_embed[:, 2]
+if umap_flag:
+    if len(args.umap) == 0:
+        umap_dim = 2
+    else:
+        umap_dim = int(args.umap[0])
+    umap_embed = umap.UMAP(n_components=umap_dim).fit_transform(Cells)
+    DRDF['umap1'] = umap_embed[:, 0]
+    DRDF['umap2'] = umap_embed[:, 1]
+    if umap_dim == 3:
+        DRDF['umap3'] = umap_embed[:, 2]
 
-(opts, args) = parser.parse_args()
-inFile = opts.inFile
-tsne_flag = opts.tsne
 
+# Prepare time-dependent color scheme
```

Comment:
I don't understand how we figure out the time corresponding to each cell. What does the ics file contain? Adding a detailed comment here will be helpful for the future.